packages/app-store/utils.ts

```diff
@@ -38,13 +40,85 @@ const credentialData = Prisma.validator<Prisma.CredentialArgs>()({
 
 export type CredentialData = Prisma.CredentialGetPayload<typeof credentialData>;
 
+export type CredentialDataWithTeamName = CredentialData & {
+  team?: {
+    name: string;
+  } | null;
+};
+
 export const ALL_APPS = Object.values(ALL_APPS_MAP);
 
-export function getLocationGroupedOptions(integrations: ReturnType<typeof getApps>, t: TFunction) {
+export async function getLocationGroupedOptions(
+  idToSearchOn: { userId: number } | { teamId: number },
+  t: TFunction
+) {
   const apps: Record<
     string,
-    { label: string; value: string; disabled?: boolean; icon?: string; slug?: string }[]
+    {
+      label: string;
+      value: string;
+      disabled?: boolean;
+      icon?: string;
+      slug?: string;
+      credential?: CredentialDataWithTeamName;
+    }[]
   > = {};
+
+  let idToSearchObject = {};
+
+  if (idToSearchOn.teamId) {
+    const teamId = idToSearchOn.teamId;
+    // See if the team event belongs to an org
+    const org = await prisma.team.findFirst({
+      where: {
+        children: {
+          some: {
+            id: teamId,
+          },
+        },
+      },
+    });
+
+    if (org) {
+      idToSearchObject = {
+        teamId: {
+          in: [teamId, org.id],
+        },
+      };
+    } else {
+      idToSearchObject = { teamId };
+    }
+  } else {
+    idToSearchObject = { userId: idToSearchOn.userId };
+  }
+
+  const credentials = await prisma.credential.findMany({
+    where: {
+      ...idToSearchObject,
```

Comment:
<img width="811" alt="Screenshot 2023-07-27 at 3 46 10 PM" src="https://github.com/calcom/cal.com/assets/1780212/96bd315f-40c1-4d64-8d81-43c288a936aa">
I think it is solvable using one prisma API call.

Reply:
Also, we are not allowing team installed apps to be available for Team member Events. Is this long term or we are going to support that?

Reply:
Nevermind: I think your approach would be much faster because you directly found out the parent teamId and then using just an **in** query.